### ISSUE TITLE
How to pre-train DAMSM model to use GPT-2 for the text encoder ?

### ISSUE THREAD

**Issue by ghost** [sidward14/Style-AttnGAN#15]
Can you give me which part of the code I need to modify and the command I should type to pre-train the DAMSM model to use GPT-2 for the text encoder? Thanks

**Comment by sidward14** [comment]
Hello! No part of the code needs to be modified.
The yaml file and the argparse input need to be changed:

First train the DAMSM to use GPT-2:
- Include `--text_encoder_type transformer` when running pretrain_DAMSM.py.

Then configure your yaml file for this DAMSM:
- Include the path to  your DAMSM trained with GPT-2 into `NET_E` in the yaml file.
- Also, the latent dim size for GPT-2 is different than biLSTM, so change `EMBEDDING_DIM` to `768` (this is written as a comment in bird_attn2_style.yml).

Finally, train your Style-AttnGAN using main.py:
- Include `--text_encoder_type transformer` when running mainy.py for training.

**Comment by ghost** [comment]
Thank you for your reply. I have another question.

In the file `cfg/DAMSM/bird.yml`, there is also `EMBEDDING_DIM`, do I need to change to `768` when pre-train the DAMSM model? Thanks.

**Comment by sidward14** [comment]
Ah yes, it'll need to be 768 there too.

**Comment by ghost** [comment]
Thank you, I have another question

Is it possible to use DAMSM model with GPT-2 text encoder for the AttnGAN model instead of Style-AttnGAN model, I want to test with the AttnGAN model. Thanks

**Comment by sidward14** [comment]
Yes, it should work. Do the same thing as above except use the AttnGAN yaml instead (e.g. bird_attn2.yml).

**Comment by ghost** [comment]
Thanks for your help.

**Comment by ghost** [comment]
Where do I need to put the `--text_encoder_type transformer` in `python pretrain_DAMSM.py --cfg cfg/DAMSM/bird.yml --gpu 0` and `python main.py --cfg cfg/bird_attn2.yml --gpu 0` ? Thanks

**Comment by sidward14** [comment]
You can put it right after the `--gpu 0`.

**Comment by ghost** [comment]
Thanks for your help.
